iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -198,7 +206,10 @@ public void configureInputJobCredentials(TableDesc tableDesc, Map<String, String
   public void configureJobConf(TableDesc tableDesc, JobConf jobConf) {
     setCommonJobConf(jobConf);
     if (tableDesc != null && tableDesc.getProperties() != null &&
-        tableDesc.getProperties().get(WRITE_KEY) != null) {
+        tableDesc.getProperties().get(InputFormatConfig.OPERATION_TYPE_PREFIX + tableDesc.getTableName()) != null) {
+      // set operation type into job conf too
+      jobConf.set(InputFormatConfig.OPERATION_TYPE_PREFIX + tableDesc.getTableName(),
+          tableDesc.getProperties().getProperty(InputFormatConfig.OPERATION_TYPE_PREFIX + tableDesc.getTableName()));
```

Comment:
nit: Maybe just here, but it took me some time to put together what is this :)
Basically:
```
String key = InputFormatConfig.OPERATION_TYPE_PREFIX + tableDesc.getTableName();
if (tableDesc.getProperties().get(key) != null) {
  jobConf.set(key, tableDesc.getProperties().get(key));
```